kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -271,25 +271,67 @@ function showExamReportDetailPage(store, classId, examId) {
   store.dispatch('CORE_SET_PAGE_LOADING', false);
 }
 
-function activateExam() {
-  this.displayModal(false);
+function activateExam(store, examId) {
+  ExamResource.getModel(examId).save({ active: true }).then(
+    () => {
+      const exams = store.state.pageState.exams;
+      const examIndex = exams.findIndex(exam => exam.id === examId);
+      exams[examIndex].active = true;
+
+      store.dispatch('SET_EXAMS', exams);
```

Comment:
While the change is in place inside the mutation, it is wrapped in the mutation, which is how vuex does its book keeping. Changes to state outside mutations are not properly tracked.